frontend/src/scenes/insights/insightDataLogic.ts

```diff
@@ -189,7 +189,11 @@ export const insightDataLogic = kea<insightDataLogicType>([
         },
         cancelChanges: () => {
             const savedFilters = values.savedInsight.filters
+            const savedResult = values.savedInsight.result
             actions.setQuery(savedFilters ? queryFromFilters(savedFilters) : null)
+            if (savedResult) {
+                actions.setInsightData({ ...values.insightData, result: values.savedInsight.result })
+            }
```

Comment:
```suggestion
actions.setInsightData({ ...values.insightData, result: savedResult ? savedResult : null })
```
We should probably reset the result in any case (wether we have data or not). Also reusing the `savedResult` var.